OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs

```diff
@@ -68,57 +84,115 @@ public void Tick(Squad owner)
 
 	class GroundUnitsAttackMoveState : GroundStateBase, IState
 	{
+		public const int StuckInPathCheckTimes = 6;
+		public const int MakeWayTick = 2;
+
+		// Give tolerance for AI grouping team at start
+		internal int StuckInPath = StuckInPathCheckTimes * 3;
+		internal int TryMakeWay = MakeWayTick;
+		internal WPos LastPos = new WPos(0, 0, 0);
+
 		public void Activate(Squad owner) { }
 
 		public void Tick(Squad owner)
 		{
+			// Basic check
 			if (!owner.IsValid)
 				return;
 
 			if (!owner.IsTargetValid)
 			{
-				var closestEnemy = FindClosestEnemy(owner);
-				if (closestEnemy != null)
-					owner.TargetActor = closestEnemy;
+				var randomSuitableEnemy = GetRandomValuableTarget(owner);
+				if (randomSuitableEnemy != null)
+					owner.TargetActor = randomSuitableEnemy;
 				else
 				{
 					owner.FuzzyStateMachine.ChangeState(owner, new GroundUnitsFleeState(), true);
 					return;
 				}
 			}
 
-			var leader = owner.Units.ClosestTo(owner.TargetActor.CenterPosition);
-			if (leader == null)
+			// Initialize pathGuide. Optimaze pathfinding by using pathGuide.
+			var pathGuide = owner.Units.FirstOrDefault();
+			if (pathGuide == null)
 				return;
 
-			var ownUnits = owner.World.FindActorsInCircle(leader.CenterPosition, WDist.FromCells(owner.Units.Count) / 3)
-				.Where(a => a.Owner == owner.Units.First().Owner && owner.Units.Contains(a)).ToHashSet();
+			// 1. Threat scan surroundings
```

Comment:
the `targetActor` here is not the one in squad so it not selecting new target, it just a check to make sure when there is enemy within range, they will convert to GroundAttackState with a valid target. 

Maybe I should change the name of  `targetActor`.

Reply:
Also `ThreatScan` here is probably confusing, it is just a check if there is enemy within a certain range. I name it to `ThreatScan` due to its checking ambush function in #17998, and it can be also reused here for tigger GroundAttackState with a valid target.

Reply:
change to scan threat